### PR TITLE
Close #38 - Feature/workflows fix

### LINKED ISSUE #38: Publish actions running in forked repositories
The three publish-related actions contained in the repository should run only on the main repository, not on the forked ones.

The owner of a forked repository could disable all actions, but in this way the Linter and CodeQL actions will be disabled too.

I propose to add an owner check in the publishing actions, so they will be skipped on forked repositories.

E.g.

```
jobs:
  publish:
    name: Publish on NuGet
    runs-on: ubuntu-latest
    if: github.repository_owner == 'marcominerva'

```

### PR DESCRIPTION
**Pull request by Defkon1**
Close #38 - Feature/workflows fix

Added owner check in publishing actions, to prevent them to run in forked repositories

### CODE REVIEW

**Comment by marcominerva** [comment]
Good catch! I'm wondering if the check can be written in a form that does not rely on a constant (the repository owner account), so that it is more portable. Something like the suggestions here: https://github.com/orgs/community/discussions/26409, but I don't know if they actually works.

**Comment by marcominerva** [comment]
OK, so I think that a little improvement to your solution could be putting the repository owner name in an environment variable (in the `env` section), so that is it easier to identify the values to change when porting the action in another repository.

**Comment by Defkon1** [comment]
If you prefer the `env` way, probably it's better to directly add a boolean-like flag `RUN_PUBLISH` and change the condition like this:

```
if: ${{ env.RUN_PUBLISH == 'true' }}
```

**Comment by Defkon1** [comment]
Yes, on a forked repo you can change whatever you prefer, this is a failsafe condition to skip actions on repositories just forked for contributions

**Comment by marcominerva** [comment]
OK, now it's clear, thanks. So, I would prefer the approach of `env` variable with the repository owner name, so that it's not "so obvious" that this variable is used to check whether to run the action or not :) Could you make this little change?

**Comment by Defkon1** [comment]
Variables are available to `job.if`, so in my last commit I changed the conditions in this way:

```
if: ${{ github.repository_owner == vars.REPOSITORY_OWNER }}
```

So actually the publishing actions are skipped if:
- there isn't a variable named `REPOSITORY_OWNER` in the repository settings
- the value of the `REPOSITORY_OWNER` variable is discordant with the actual repository owner

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
   publish:
     name: Publish on NuGet
     runs-on: ubuntu-latest
-    if: ${{ github.repository_owner == env.REPOSITORY_OWNER }}
+    if: ${{ github.repository_owner == vars.REPOSITORY_OWNER }}
```
Comment (marcominerva) [inline .github/workflows/publish.yml:19]: According to https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables, the variable name should be GITHUB_REPOSITORY_OWNER:

image

Moreover, are you sure that this information is accessible also trough `vars`?

**Review by marcominerva — CHANGES_REQUESTED** [review]

**Pull request merged**